nexus/types/src/quiesce.rs

```diff
@@ -138,21 +156,53 @@ struct SagaQuiesceInner {
     /// given reassignment pass.  See `reassignment_done()` for details.
     recovered_reassignment_generation: Generation,
 
-    /// whether a saga recovery operation is ongoing, and if one is, what
-    /// `reassignment_generation` was when it started
-    recovery_pending: Option<Generation>,
+    /// blueprint id that saga recovery has "caught up to"
+    ///
+    /// This means that we have finished recovering any sagas that were
+    /// re-assigned to us due to expungements of other Nexus zones up through
+    /// this blueprint.  Put differently: we know that we will never be assigned
+    /// more sagas due to expungement unless the target blueprint changes past
+    /// this one.
+    ///
+    /// This does not mean that we've fully drained all sagas up through this
+    /// blueprint.  There may still be sagas running.
+    recovered_blueprint_id: Option<BlueprintUuid>,
+
+    /// blueprint id that we're "fully drained up to"
+    ///
+    /// If this value is non-`None`, that means that:
+    ///
+    /// - saga creation is disallowed
+    /// - no sagas are running
+    /// - we have re-assigned sagas from other Nexus instances expunged in this
+    ///   blueprint or earlier
+    /// - we have finished recovery for all those sagas (that had been assigned
+    ///   to us as of the re-assignment pass for this blueprint id)
+    ///
+    /// This means that the only way we can wind up running another saga is if
+    /// there's a new blueprint that expunges a different Nexus zone.
+    drained_blueprint_id: Option<BlueprintUuid>,
+
+    /// whether a saga recovery operation is ongoing, and if one is:
+    /// - what `reassignment_generation` was when it started
+    /// - which blueprint id we'll be fully caught up to upon completion
+    #[serde(skip)] // XXX-dap
```

Comment:
`XXX` here because we don't want to skip this field?

Reply:
Yes -- sorry I missed that!  This is a problem because we don't support a tuple in this context in the OpenAPI spec.  I will replace it with a struct.